### ISSUE TITLE
Base path

### ISSUE THREAD

**Issue by phanicode** [CARTAvis/carta-backend-ICD-test#254]
What does "base": "$BASE" mean in con here, the base path of the linux system?
https://github.com/CARTAvis/carta-backend-ICD-test/blob/623a4c42ff44621eadcdc260c4ac310944e32bd5/src/test/config.json#L7

Does this mean that the backend has to search the entire system or the cartadata directory recursively?

**Comment by acdo2002** [comment]
No. The ICD tests use `config.serverURL` to connect to the server.

If you would like the ICD tests run on your local system. 
You have to build the [carta-backend](https://github.com/CARTAvis/carta-backend) first and set it up.

Then change each carta-backend-ICD-test/src/test/*.test.ts 
from `let testServerUrl = config.serverURL` to `let testServerUrl = config.serverURL0;`
It should connect to the directory you have set up in the backend.

**Comment by phanicode** [comment]
Let me rephrase my question

1) Is the FILE_LIST_RESPONSE on the backend supposed to recursively search for files?

2) If so, why is "aJ.fits" [  inside set_QA of the repo](https://github.com/CARTAvis/carta-backend-ICD-test/tree/master/public/set_QA)   
whereas 
[in this test](https://github.com/CARTAvis/carta-backend-ICD-test/blob/68c5ee04c56c642bde70156d163db0e98f1c05a0/src/test/GET_FILELIST.test.ts#L33)

set_QA is supposed to be a sub directory of the directory containing aJ.fits
Understanding these tests are relevant for me as I'm working on a python CARTA backend 

@acdo2002

**Comment by acdo2002** [comment]
1. <== FILE_INFO_RESPONSE only returns when FILE_LIST_REQUEST ==> has been set.
one FILE_LIST_REQUEST ==>  corresponds to one <== FILE_INFO_RESPONSE. 
Except the folder needs more time to open, the ICD messages show sequence FILE_LIST_PROGRESS messages

If you would like to understand how ICD interacts with the backend.
you can check: [demoserver](http://carta.asiaa.sinica.edu.tw/frontend/dev/?socketUrl=wss://carta.asiaa.sinica.edu.tw/socketdev)
Let console log can display the ICD messages: File -> Preferences -> Log Events -> Select all

You can open any file (e.g. set_QA/M17_SWex.fits), then open Developer tools/Console (chrome browser) and check ICD messages.

2. We just put all test files inside the `public` directory together with carta-backend-ICD-test repository.
here is my example, if you set `/home/mingyi/carta-test-img/` as your folder to open with the carta_backend
(e.g. `./carta_backend /home/mingyi/carta-test-img --debug_no_auth=true --no_http=true --verbosity=5 --log_performance=true --omp_threads=4 --top_level_folder=/home/mingyi/carta-test-img --frontend_folder=/home/mingyi/carta-test-img`)

the structure inside `/home/mingyi/carta-test-img/` should like
```
-/home/mingyi/carta-test-img/
  -aJ.fits
  -set_QA
    -M17_SWex.fits
    -empty_folder
```
Hope this reply can answer your question, let me know if you had further question!
